src/prototype_room_my.js

```diff
@@ -326,6 +326,57 @@ Room.prototype.executeRoom = function() {
     }
   }
 
+  let carryHelpInterval = config.carryHelpers.ticksUntilHelpCheck;
+  let key = _.findKey(Memory.needEnergyRooms, this.name);
+
+  if (Game.time % carryHelpInterval) {
+    this.memory.energyAvailableSum += this.energyAvailable;
```

Comment:
You could work with a moving average: `this.memory.energyAvailable = (1 - factor) * this.memory.energyAvailable + (factor) this.energyAvailable`. E.g. factor 0.9 means, that the current value is very important, but take 0.1 from the previous moving average.
I think I would do something like 0.2. This way you have a better variable in memory and IMHO less complexity.